### ISSUE TITLE
signal_strength = true in Net block , causes graphical glitch...

### ISSUE THREAD

**Issue by ReductoAdAbsurdum** [greshake/i3status-rust#467]
when I set `signal_strength = true `  in the Net block and reload i3, just before signal strength loads I see black spots on the status line in the net area which disappears  after signal strength is fully loaded....

**Comment by GladOSkar** [comment]
Can confirm:

Before load:
![image](https://user-images.githubusercontent.com/9497832/68933367-cd59f380-0794-11ea-85f8-c49a6d2085f4.png)

After Load:
![image](https://user-images.githubusercontent.com/9497832/68933391-d8148880-0794-11ea-8a3f-3da16d1592d8.png)

**Comment by ifreund** [comment]
You can also reproduce this by disabling everything in the net block except for the signal strength icon.  The graphical bug then persists forever.

**Comment by ammgws** [comment]
Can someone using `sway` confirm whether they can reproduce this?

Also, can someone who's experiencing this run i3status-rs from the command line and post the output? Interested to see if there's something strange there.

**Comment by ammgws** [comment]
![image](https://user-images.githubusercontent.com/20397027/80814994-d4bc6e80-8c07-11ea-96c4-23c8737f3f7f.png)

I can't reproduce on sway. I'm using a simulated wireless device but it shouldn't make a difference... :/

**Comment by ammgws** [comment]
Closing due to lack of response and not being able to reproduce it here. Please re-open if you have further info.

**Comment by GladOSkar** [comment]
I just checked and the block does actually send the wrong color information before it's loaded:

Before loading:
```
{
    "background":"#000000",
    "color":"#000000",
    "full_text":"",
    "markup":"pango",
    "separator":false,
    "separator_block_width":0
}
```

After loading:
```
{
    "background":"#535353FF",
    "color":"#FFFFFFFF",
    "full_text":" 84% ",
    "markup":"pango",
    "name":"b74fa0319ac2436086692d7fa59e7d60",
    "separator":false,
    "separator_block_width":0
}
```

So this probably is actually an issue that is solveable in code

**Comment by ammgws** [comment]
Is it just the net block that has that behaviour?

**Comment by GladOSkar** [comment]
It's the only one i ever see it on, yes. All other blocks have their colors set correctly immediately.

Also, it's worth noting that this only happens for the signal strength widget, not the entire block.

Unfortunately i'm not well-versed enough in rust to debug this. It also doesn't bother me that much since it's only ever there for a couple of seconds.

**Comment by ammgws** [comment]
Does it happen when using the recently introduced `format` string for the net block?

**Comment by GladOSkar** [comment]
I'll have a look

**Comment by ammgws** [comment]
IIRC it will still use widgets if the deprecated options are used, so make sure only the format option is specified in your config when you try.

**Comment by GladOSkar** [comment]
First of all unrelatedly, after updating my current config doesn't work anymore. Even though i have set

```
[[block]]
block = "net"
device = "ppp0"
hide_inactive = true
hide_missing = true
interval = 10
speed_up = false
speed_down = false
ip = false

[[block]]
block = "net"
device = "eno1"
hide_inactive = true
hide_missing = true
interval = 10
speed_up = false
speed_down = false
ip = false

[[block]]
block = "net"
device = "wlp2s0"
signal_strength = true
interval = 10
speed_up = false
speed_down = false
ip = false
```

The blocks show their speeds:
![image](https://user-images.githubusercontent.com/9497832/86793146-78445800-c06b-11ea-9dc9-1b02dbf876c0.png)

This is probably unintended and should be fixed?

**Comment by GladOSkar** [comment]
Alright, with the format string it not shows this before initialization, which is way better:

![image](https://user-images.githubusercontent.com/9497832/86793570-e38e2a00-c06b-11ea-83b7-bf2933f15b00.png)

**Comment by ammgws** [comment]
Resolved via #738
